.github/workflows/hlsl-macos.yaml

```diff
@@ -0,0 +1,26 @@
+name: HLSL Tests - macOS
+
+permissions:
+  contents: read
+  checks: write
```

Comment:
Why do you need `checks: write` here ?

Reply:
Probably don't, I think this was a misunderstanding on my part about the permissions required for the action to report its status check. Will remove.